mle/agents/advisor.py

```diff
@@ -102,6 +103,14 @@ def __init__(self, model, console=None):
         self.sys_prompt += self.json_mode_prompt
         self.chat_history.append({"role": 'system', "content": self.sys_prompt})
 
+    def clean_json_string(self, input_string):
+        cleaned = input_string.strip()
+        cleaned = re.sub(r'^```\s*json?\s*', '', cleaned)
+        cleaned = re.sub(r'\s*```\s*$', '', cleaned)
+        print("Cleaned:",cleaned)
+        parsed_json = json.loads(cleaned)
+        return parsed_json
```

Comment:
Many thanks! It appears that Mistral now officially supports `response_format='json'` (https://docs.mistral.ai/capabilities/json_mode/), so is it still necessary to manually clean the JSON string?

Reply:
So the code here is already using that - https://github.com/MLSysOps/MLE-agent/blob/13968d7c79807ebcd0a95e3fdb7d23b06805dcbb/mle/agents/advisor.py#L117 - but the model is still not returning only json.